pkgs/applications/networking/sync/rclone/default.nix

```diff
@@ -19,6 +19,9 @@ buildGoPackage rec {
 
   postInstall = ''
     install -D -m644 $src/rclone.1 $man/share/man/man1/rclone.1
+    mkdir -p $bin/share/zsh/site-functions/ $bin/share/bash-completion/completions/
+    $NIX_BUILD_TOP/go/bin/rclone genautocomplete zsh $bin/share/zsh/site-functions/_rclone
```

Comment:
This has the same issue with cross-compilation. `$NIX_BUILD_TOP/go/bin/rclone` is the compiled for the host platform and wouldn't run on the build platform. Building ` pkgsCross.aarch64-multiplatform.rclone` produces this error:

```
/nix/store/7hqfwijjqda0p0ppdhm0b6i3xg5blgnr-stdenv-linux/setup: line 83: /build/go/bin/rclone: cannot execute binary file: Exec format error
```
I think you should conditionally use `buildPackages.rclone` if `stdenv.buildPlatform  != stdenv.hostPlatform` like in the earlier suggested changes. For reference, note the similar code in `pkgs/shells/zsh/default.nix` for generating autocomplete files in a cross-compatible way.

You can test with `nix-build . -A pkgsCross.aarch64-multiplatform.rclone`--which doesn't take too long since most dependencies are already in the cache.

Reply:
Sorry for the late response, I've applied your original suggestion